src/main/java/com/ood/myorange/util/StorageUtilFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.ood.myorange.util;
+
+import com.ood.myorange.config.storage.S3Configuration;
+import com.ood.myorange.pojo.StorageConfig;
+import com.ood.myorange.util.storageUtils.AWSUtil;
+import com.ood.myorange.util.storageUtils.StorageUtil;
+import com.ood.myorange.config.storage.StorageConfiguration;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class StorageUtilFactory {
+    private static HashMap<StorageConfiguration.StorageType, StorageUtil> storageUtilHashMap =
```

Comment:
What if we have two S3? just like you mentioned in `testdata.sql`

Reply:
We probably will only have one config for every storage provider.